src/base/Identifier.ml

```diff
@@ -19,21 +19,89 @@
 open Core_kernel
 open ErrorUtils
 
-type 'rep t = Ident of string * 'rep [@@deriving sexp]
+module type QualifiedName = sig
 
-let asId i = Ident (i, dummy_loc)
+  type t [@@deriving sexp]
 
-let asIdL i loc = Ident (i, loc)
+  val as_string : t -> string
 
-let get_id i = match i with Ident (x, _) -> x
+  val as_error_string : t -> string
 
-let get_rep i = match i with Ident (_, l) -> l
+  val equal_id : t -> t -> bool
+  
+  val compare_id : t -> t -> int
+    
+end
+
+module FlattenedName = struct
+
+  type t = string [@@deriving sexp]
+
+  let as_string n = n
+
+  let as_error_string n = n
+
+  let equal_id = String.(=)
+
+  let compare_id = String.compare
+
+end
+
+
+module type ScillaIdentifier = sig
+
+  module Name : QualifiedName
+    
+  type 'rep t = Ident of Name.t * 'rep [@@deriving sexp]
+
+  val asId : Name.t -> loc t
+
+  val asIdL : Name.t -> 'a -> 'a t
```

Comment:
Perhaps we could rename these into something like `mk_loc` and `mk`? (or something like that)

Reply:
Probably `mk_loc_id` and `mk_id`, but sure.

Reply:
So one is supposed to to always open the namespace when using these constructors?
It's just if it should be a qualified name, then `Identifier.mk` looks less redundant than `Identifier.mk_id` :)

Reply:
I'm not sure what you mean.

Once the new pass has been run, then there is no need to inspect the names again, but we still need to construct new identifiers and annotations out of old identifiers and annotations. We could have an `add_annot` function, but that belongs in the `Rep` modules rather than in the `Identifier` module.

Reply:
ok, nevermind :)